src/main/webapp/resources/js/pages/launch/launch-context.js

```diff
@@ -0,0 +1,59 @@
+import React from "react";
+import { getPipelineDetails } from "../../apis/pipelines/pipelines";
+
+const LaunchStateContext = React.createContext();
+const LaunchDispatchContext = React.createContext();
+
+const TYPES = {
+  LOADED: "launch:loaded",
+};
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case TYPES.LOADED:
+      return { ...state, loading: false, ...action.details };
+  }
+};
+
+function LaunchProvider({ children }) {
+  const [id] = React.useState(() => {
+    const params = new URLSearchParams(window.location.search);
+    return params.get("id");
+  });
+
+  const [state, dispatch] = React.useReducer(reducer, { loading: true });
+
+  React.useEffect(() => {
+    getPipelineDetails({ id }).then((data) => {
+      const { name, description, ...details } = data;
+      details.pipeline = { name, description };
+      dispatch({ type: TYPES.LOADED, details });
+    });
+  }, [id]);
+
+  return (
+    <LaunchStateContext.Provider value={state}>
+      <LaunchDispatchContext.Provider value={dispatch}>
+        {children}
+      </LaunchDispatchContext.Provider>
+    </LaunchStateContext.Provider>
+  );
+}
+
+function useLaunchState() {
```

Comment:
Can you add a comment to this function and the one below explaining when they are to be used

Reply:
LOL, sure.  But if you read the article I sent you you would know :)

Reply:
LOL still need to read it :)